fiona/__init__.py

```diff
@@ -77,6 +77,17 @@
     class Path:
         pass
 
+
+# Add gdal dll directory on Windows and Python >= 3.8, see https://github.com/Toblerity/Fiona/issues/851
+if platform.system() == 'Windows' and (3, 8) <= sys.version_info:
```

Comment:
is add_dll_directory harmful when gdal dlls are already loaded? E.g. when multiple gdal binaries are present.